src/lib/IFrameCommunicatorTypes.ts

```diff
@@ -0,0 +1,38 @@
+export interface IPluginInitializedMessageData {
+  readonly projectId: string | null;
+  readonly languageCodename: string | null;
+  readonly enabled: boolean;
+}
+
+export interface IElementClickedMessageMetadata {
+  readonly elementRect: DOMRect;
+}
+
+export interface IElementClickedMessageData {
+  readonly projectId: string;
+  readonly languageCodename: string;
+  readonly itemId: string;
+  readonly elementCodename: string;
+  readonly contentComponentId: string;
+}
+
+export interface IPluginStatusMessageData {
+  readonly enabled: boolean;
+}
+
+export enum IFrameMessageType {
+  ElementDummy = 'kontent-smart-link:element:dummy',
+  ElementClicked = 'kontent-smart-link:element:clicked',
+  ElementRemoveRequest = 'kontent-smart-link:element:remove:request',
+  Initialized = 'kontent-smart-link:initialized',
+  Status = 'kontent-smart-link:status',
+}
+
+export enum IFrameResponseType {
+  ElementRemoveRequestResponse = 'element-remove-request-response',
```

Comment:
is this remove response here simply for having an example? Otherwise there's no need to have any remove requests/response since it is not going to be part of this iterations scope.
If you wanted to show some example, then I'd use the `dummy` as before.